api/utils/sshutils/chconn.go

```diff
@@ -38,24 +38,25 @@ type Conn interface {
 
 // NewChConn returns a new net.Conn implemented over
 // SSH channel
-func NewChConn(conn Conn, ch ssh.Channel) *ChConn {
-	return newChConn(conn, ch, false)
+func NewChConn(conn Conn, ch ssh.Channel, remoteAddr net.Addr) *ChConn {
+	return newChConn(conn, ch, false, remoteAddr)
 }
 
 // NewExclusiveChConn returns a new net.Conn implemented over
 // SSH channel, whenever this connection closes
-func NewExclusiveChConn(conn Conn, ch ssh.Channel) *ChConn {
-	return newChConn(conn, ch, true)
+func NewExclusiveChConn(conn Conn, ch ssh.Channel, remoteAddr net.Addr) *ChConn {
+	return newChConn(conn, ch, true, remoteAddr)
```

Comment:
That's how I originally implemented it. Edoardo suggested since we already control this type we can just put it in there and that made sense to me. But I'm not tied to any of the two implementations, so if you really think it's better to just wrap I can revert it.

Reply:
I've reverted the commit to restore implementation with wrapped connection, later we can see if overriding it inside `ChConn` will become more useful.